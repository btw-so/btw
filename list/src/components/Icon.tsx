import React from 'react';
import SVG from 'react-inlinesvg';
import styled from 'styled-components';
import { px } from 'styled-minimal';

import { Icons } from 'types';

interface Props {
  name: Icons;
  width?: number;
}

const IconWrapper = styled(SVG)`
  display: inline-block;
  line-height: 0;

  svg {
    height: auto;
    max-height: 100%;
    width: ${({ width }) => width};
  }
`;

function Icon({ name, width = 20 }: Props) {
  return (
    <IconWrapper
      height="100%"
      src={`${process.env.PUBLIC_URL}/media/icons/${name}.svg`}
      width={px(width)}
    />
  );
}

export default Icon;
