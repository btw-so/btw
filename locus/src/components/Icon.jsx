import React from 'react';
import SVG from 'react-inlinesvg';
import styled from 'styled-components';
import { px } from 'styled-minimal';

// import { Icons } from 'types'; // Removed TypeScript type import

// Removed TypeScript interface Props
// function Icon({ name, width = 20 }: Props) {
function Icon({ name, width = 20 }) {
  return (
    <IconWrapper
      height="100%"
      src={`${process.env.PUBLIC_URL}/media/icons/${name}.svg`}
      width={px(width)}
    />
  );
}

const IconWrapper = styled(SVG)`
  display: inline-block;
  line-height: 0;

  svg {
    height: auto;
    max-height: 100%;
    width: ${({ width }) => width};
  }
`;

export default Icon;
