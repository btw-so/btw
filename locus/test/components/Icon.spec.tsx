import React from 'react';

import Icon from 'components/Icon';

import { render } from 'test-utils';

describe('Icon', () => {
  it('should render properly', () => {
    const { container } = render(<Icon name="check" />);

    expect(container).toMatchSnapshot();
  });
});
