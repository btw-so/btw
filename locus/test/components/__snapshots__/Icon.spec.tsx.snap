// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render properly 1`] = `
.c0 {
  display: inline-block;
  line-height: 0;
}

.c0 svg {
  height: auto;
  max-height: 100%;
  width: 20px;
}

<div>
  <svg
    class="c0"
    height="100%"
    src="/media/icons/check.svg"
    width="20px"
  />
</div>
`;
